x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/saml/SamlAttributes.java

```diff
@@ -103,4 +151,41 @@ public String toString() {
         }
     }
 
+    static class SamlSecureAttribute extends AbstractSamlAttribute<SecureString> implements Releasable {
+
+        SamlSecureAttribute(Attribute attribute) {
+            super(
+                attribute.getName(),
+                attribute.getFriendlyName(),
+                attribute.getAttributeValues()
+                    .stream()
+                    .map(x -> x.getDOM().getTextContent())
+                    .filter(Objects::nonNull)
+                    .map(String::toCharArray)
+                    .map(SecureString::new)
+                    .toList()
+            );
+        }
+
+        SamlSecureAttribute(String name, @Nullable String friendlyName, List<SecureString> values) {
+            super(name, friendlyName, values);
+        }
+
+        @Override
+        public String toString() {
+            StringBuilder str = new StringBuilder();
+            if (Strings.isNullOrEmpty(friendlyName)) {
+                str.append(name);
+            } else {
+                str.append(friendlyName).append('(').append(name).append(')');
+            }
+            str.append("=").append("[::es_redacted::]").append("(len=").append(values.size()).append(')');
```

Comment:
Why the `es_redacted`? Can't we just skip it entirely?
```suggestion
            str.append("[").append(values.size()).append(" value(s)]');
```